src/binary_tree.rs

```diff
@@ -383,6 +385,20 @@ impl<C: Mergeable> MatchedPair<C> {
     }
 }
 
+impl<C: Clone> From<Vec<Node<C>>> for MatchedPair<C> {
```

Comment:
The point here is to
a) take ownership of the parameters, not clone
b) check that the left node lives exactly to the left of the right node, otherwise panic

Input should be 2 nodes: left & right, not a vector. And also need to swap constructions of the form `MatchedPair {left, right}` with `MatchedPair::from(left, right)`.

Reply:
Cool.

`from()` only takes a single argument, that's why I used a `Vec`.

Reply:
Got you. Use a tuple then.